### PR TITLE
Deprecate our SimpleXmlConverterFactory

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: retrofit-converters/simplexml/src/main/java/retrofit2/converter/simplexml/SimpleXmlConverterFactory.java [hunk 1]
```diff
@@ -29,6 +29,8 @@
  * <p>
  * This converter only applies for class types. Parameterized types (e.g., {@code List<Foo>}) are
  * not handled.
+ *
+ * @deprecated we recommend switching to the JAXB converter.
  */
 public final class SimpleXmlConverterFactory extends Converter.Factory {
```
Comment (JakeWharton) [inline retrofit-converters/simplexml/src/main/java/retrofit2/converter/simplexml/SimpleXmlConverterFactory.java]: Need `@Deprecated` or the bit won't be set in the class file and classfile-only consumers (almost everyone) won't see it as deprecated.

Comment (swankjesse) [reply retrofit-converters/simplexml/src/main/java/retrofit2/converter/simplexml/SimpleXmlConverterFactory.java]: Untrue.

Comment (swankjesse) [reply retrofit-converters/simplexml/src/main/java/retrofit2/converter/simplexml/SimpleXmlConverterFactory.java]: https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.7.15

Comment (swankjesse) [reply retrofit-converters/simplexml/src/main/java/retrofit2/converter/simplexml/SimpleXmlConverterFactory.java]: Fixed regardless!

Comment (JakeWharton) [reply retrofit-converters/simplexml/src/main/java/retrofit2/converter/simplexml/SimpleXmlConverterFactory.java]: That link refers to the bit. I don't see anything about whether the Javadoc tag sets it.

Comment (JakeWharton) [reply retrofit-converters/simplexml/src/main/java/retrofit2/converter/simplexml/SimpleXmlConverterFactory.java]: Tests show that it does, though.

Comment (swankjesse) [reply retrofit-converters/simplexml/src/main/java/retrofit2/converter/simplexml/SimpleXmlConverterFactory.java]: Yeah. Java had `@deprecated` long before it had annotations. Having both it and`@Deprecated` is redundant (but common!).

**Pull request merged**